src/main/java/algorithms/medium/Permutations2.java

```diff
@@ -0,0 +1,44 @@
+package algorithms.medium;
+
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+
+public class Permutations2 {
+
+    public List<List<Integer>> permuteUnique(int[] nums) {
+        List<List<Integer>> output = new ArrayList<List<Integer>>();
+        backtrack(0, nums, output);
+        return output;
+
+    }
+    
+    public void backtrack(int choiceIndex, int[] nums, List<List<Integer>> output) {
+        if (choiceIndex == nums.length) {
+            ArrayList<Integer> permutation = new ArrayList<Integer>();
+            for (int num : nums) {
+                permutation.add(num);
+            }
+            output.add(permutation);
+        }
+
+        Set<Integer> visited = new HashSet<>();
+        for (int i = choiceIndex; i < nums.length; i++) {
+            if (!visited.add(nums[i])) {
+                continue;
+            }
```

Comment:
```suggestion
        boolean[] visited = new boolean[21];
        for (int i = choiceIndex; i < nums.length; i++) {
            if (visited[nums[i] + 10]) {
                continue;
            }
            visited[nums[i] + 10] = true;
```

Reply:
Which is what we use in the palindrome permutation solution

Reply:
Nice idea Erdem. Thanks. Suggestion commited.